### ISSUE TITLE
Proximity SDK crashes while running and user removed location permission

### ISSUE THREAD

**Issue by GraceHuntley** [Estimote/Android-Proximity-SDK#51]
## Prerequisites

* [x ] Put an X between the brackets on this line if you have done all of the following:
    * My beacons have Estimote Monitoring enabled (Check it in the Estimote Cloud, or via app)
    * My Android device/devices supports BLE and has Android OS version >= 5.0.0 
    * My Android device/devices have bluetooth enabled
    * My app has Location Permissions granted
    
## Basic information

**Estimote SDK version:** [Paste here the SDK version that you are using (like `0.1.0-alpha.6`)]
0.5.2

**Android devices affected:** [Paste here your Android device model/models that you are having problems with.]
All

**Android OS version affected:** [Android OS version (ex. `Oreo 8.0.0`)]
All
**Beacon hardware version:** [Paste here your beacons hardware version (you can check that in Estimote Cloud).] 

## Description

When estimate is running in background and a user revokes location permission sdk crashes app.

`FATAL EXCEPTION: main
                                                                            Process: com.visa.superbowl.concierge, PID: 3676
                                                                            java.lang.RuntimeException: Unable to start service com.estimote.scanning_plugin.packet_provider.service.PacketProviderWrapperService@1479c86 with null: kotlin.UninitializedPropertyAccessException: lateinit property serviceHelper has not been initialized
                                                                                at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3694)
                                                                                at android.app.ActivityThread.-wrap21(Unknown Source:0)
                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1801)
                                                                                at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                                at android.os.Looper.loop(Looper.java:164)
                                                                                at android.app.ActivityThread.main(ActivityThread.java:6938)
                                                                                at java.lang.reflect.Method.invoke(Native Method)
                                                                                at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
                                                                             Caused by: kotlin.UninitializedPropertyAccessException: lateinit property serviceHelper has not been initialized
                                                                                at com.estimote.scanning_plugin.packet_provider.service.PacketProviderWrapperService.showNotification(PacketProviderWrapperService.kt:82)
                                                                                at com.estimote.scanning_plugin.packet_provider.service.PacketProviderWrapperService.onStartCommand(PacketProviderWrapperService.kt:32)
                                                                                at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3677)
                                                                                at android.app.ActivityThread.-wrap21(Unknown Source:0) 
                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1801) 
                                                                                at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                                                at android.os.Looper.loop(Looper.java:164) 
                                                                                at android.app.ActivityThread.main(ActivityThread.java:6938) 
                                                                                at java.lang.reflect.Method.invoke(Native Method) 
                                                                                at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327) 
                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374) 
05-11 10:46:48.202 3676-3700/com.visa.superbowl.concierge E/GAv4: Successfully bound to service but never got onServiceConnected callback`

**Comment by pawelDylag** [comment]
mnworiugj]w-L;LB

**Comment by pawelDylag** [comment]
Hello @GraceHuntley 

Thank you very much for reporting this. This is definitely a bug, and we are able to reproduce it. There will be a fix for this in the upcoming releases :)

Here is a cookie for you -> 🍪 

Cheers, 
Paweł

**Comment by TeepsSmith** [comment]
Hey @GraceHuntley  @pawelDylag  I am also having this bug. Is this still being worked on?
I noticed it if I had continued into the app after denying the request to enable my BT then later turned it on manually and then each subsequent app launch without uninstalling.

**Comment by Goncalo-Almeida** [comment]
Hi, I'm also getting this error.
I'm getting this on a OnePlus with Android 8.

Let me know if I can help with anything.
I'm attaching the stacktrace and info from Crashlytics.
``` 
### OS Version: 8.0.0
### Device: ONEPLUS A3003
### RAM Free: 35.9%
### Disk Free: 41.5%

#0. Crashed: main
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3721)
       at android.app.ActivityThread.-wrap20(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1793)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6798)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

--

Fatal Exception: java.lang.RuntimeException: Unable to start service com.estimote.scanning_plugin.packet_provider.service.PacketProviderWrapperService@71e3ddc with null: kotlin.UninitializedPropertyAccessException: lateinit property serviceHelper has not been initialized
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3721)
       at android.app.ActivityThread.-wrap20(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1793)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6798)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

Caused by kotlin.UninitializedPropertyAccessException: lateinit property serviceHelper has not been initialized
       at com.estimote.scanning_plugin.packet_provider.service.PacketProviderWrapperService.showNotification(Unknown Source:6)
       at com.estimote.scanning_plugin.packet_provider.service.PacketProviderWrapperService.onStartCommand(Unknown Source:3)
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3704)
       at android.app.ActivityThread.-wrap20(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1793)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6798)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

#0. Crashed: main
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3721)
       at android.app.ActivityThread.-wrap20(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1793)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6798)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

#1. OkHttp ConnectionPool
       at java.lang.Object.wait(Object.java)
       at com.android.okhttp.ConnectionPool$1.run(ConnectionPool.java:103)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

#2. Thread-8
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2101)
       at java.util.concurrent.LinkedBlockingQueue.poll(LinkedBlockingQueue.java:467)
       at com.google.android.gms.common.BlockingServiceConnection.getServiceWithTimeout(Unknown Source:22)
       at com.google.android.gms.ads.identifier.AdvertisingIdClient.zza(Unknown Source:4)
       at com.google.android.gms.ads.identifier.AdvertisingIdClient.zza(Unknown Source:27)
       at com.google.android.gms.ads.identifier.AdvertisingIdClient.getAdvertisingIdInfo(Unknown Source:47)
       at java.lang.reflect.Method.invoke(Method.java)
       at io.fabric.sdk.android.services.common.AdvertisingInfoReflectionStrategy.getInfo(Unknown Source:27)
       at io.fabric.sdk.android.services.common.AdvertisingInfoReflectionStrategy.getAdvertisingId(Unknown Source:15)
       at io.fabric.sdk.android.services.common.AdvertisingInfoReflectionStrategy.getAdvertisingInfo(Unknown Source:10)
       at io.fabric.sdk.android.services.common.AdvertisingInfoProvider.getAdvertisingInfoFromStrategies(Unknown Source:4)
       at io.fabric.sdk.android.services.common.AdvertisingInfoProvider.getInfoFromPreferences(Unknown Source)
       at io.fabric.sdk.android.services.common.AdvertisingInfoProvider$1.onRun(Unknown Source:2)
       at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(Unknown Source:5)
       at java.lang.Thread.run(Thread.java:764)

#3. Queue
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
       at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.performOperation(Unknown Source:24)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.get(Unknown Source)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source:2)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1087)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1147)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

#4. DBFlow Transaction Queue
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
       at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
       at com.raizlabs.android.dbflow.structure.database.transaction.DefaultTransactionQueue.run(Unknown Source:10)

#5. TcmReceiver
       at android.net.LocalSocketImpl.readba_native(LocalSocketImpl.java)
       at android.net.LocalSocketImpl.-wrap1(Unknown Source)
       at android.net.LocalSocketImpl$SocketInputStream.read(LocalSocketImpl.java:110)
       at com.qti.tcmclient.DpmTcmClient$TcmReceiver.run(DpmTcmClient.java:143)
       at java.lang.Thread.run(Thread.java:764)

#6. FinalizerWatchdogDaemon
       at java.lang.Object.wait(Object.java)
       at java.lang.Daemons$FinalizerWatchdogDaemon.sleepUntilNeeded(Daemons.java:310)
       at java.lang.Daemons$FinalizerWatchdogDaemon.runInternal(Daemons.java:285)
       at java.lang.Daemons$Daemon.run(Daemons.java:111)
       at java.lang.Thread.run(Thread.java:764)

#7. Crashlytics Exception Handler1
       at dalvik.system.VMStack.getThreadStackTrace(VMStack.java)
       at java.lang.Thread.getStackTrace(Thread.java:1536)
       at java.lang.Thread.getAllStackTraces(Thread.java:1586)
       at com.crashlytics.android.core.CrashlyticsController.writeSessionEvent(Unknown Source:102)
       at com.crashlytics.android.core.CrashlyticsController.writeFatal(Unknown Source:78)
       at com.crashlytics.android.core.CrashlyticsController.enableExceptionHandling(Unknown Source)
       at com.crashlytics.android.core.CrashlyticsController$7.call(Unknown Source:17)
       at com.crashlytics.android.core.CrashlyticsController$7.call(Unknown Source)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at io.fabric.sdk.android.services.common.ExecutorUtils$1$1.onRun(Unknown Source:2)
       at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(Unknown Source:5)
       at java.lang.Thread.run(Thread.java:764)

#8. main
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
       at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:447)
       at java.util.concurrent.FutureTask.get(FutureTask.java:205)
       at com.crashlytics.android.core.CrashlyticsBackgroundWorker.submitAndWait(Unknown Source:21)
       at com.crashlytics.android.core.CrashlyticsController.enableExceptionHandling(Unknown Source:62)
       at com.crashlytics.android.core.CrashlyticsController$6.onUncaughtException(Unknown Source:2)
       at com.crashlytics.android.core.CrashlyticsUncaughtExceptionHandler.uncaughtException(Unknown Source:13)
       at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
       at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1063)
       at java.lang.Thread.dispatchUncaughtException(Thread.java:1953)

#9. Queue
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
       at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.performOperation(Unknown Source:24)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.get(Unknown Source)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source:2)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1087)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1147)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

#10. Queue
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
       at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.performOperation(Unknown Source:24)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.get(Unknown Source)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source:2)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1087)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1147)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

#11. Answers Events Handler1
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2101)
       at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1132)
       at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1087)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1147)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at io.fabric.sdk.android.services.common.ExecutorUtils$1$1.onRun(Unknown Source:2)
       at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(Unknown Source:5)
       at java.lang.Thread.run(Thread.java:764)

#12. Queue
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
       at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:450)
       at java.util.concurrent.FutureTask.get(FutureTask.java:192)
       at com.crashlytics.android.core.CrashlyticsBackgroundWorker.submitAndWait(Unknown Source:32)
       at com.crashlytics.android.core.CrashlyticsController.enableExceptionHandling(Unknown Source:7)
       at com.crashlytics.android.core.CrashlyticsCore.doInBackground(Unknown Source:94)
       at com.crashlytics.android.core.CrashlyticsCore.doInBackground(Unknown Source)
       at io.fabric.sdk.android.InitializationTask.onPreExecute(Unknown Source:14)
       at io.fabric.sdk.android.InitializationTask.onPreExecute(Unknown Source:2)
       at io.fabric.sdk.android.services.concurrency.AsyncTask$2.call(Unknown Source:21)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

#13. ReferenceQueueDaemon
       at java.lang.Object.wait(Object.java)
       at java.lang.Daemons$ReferenceQueueDaemon.runInternal(Daemons.java:186)
       at java.lang.Daemons$Daemon.run(Daemons.java:111)
       at java.lang.Thread.run(Thread.java:764)

#14. Queue
       at java.lang.Object.wait(Object.java)
       at java.lang.Thread.parkFor$(Thread.java:2135)
       at sun.misc.Unsafe.park(Unsafe.java:358)
       at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
       at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.performOperation(Unknown Source:24)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.get(Unknown Source)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source:2)
       at io.fabric.sdk.android.services.concurrency.DependencyPriorityBlockingQueue.take(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1087)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1147)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

#15. FinalizerDaemon
       at java.lang.Object.wait(Object.java)
       at java.lang.Object.wait(Object.java:422)
       at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:188)
       at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:209)
       at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:240)
       at java.lang.Daemons$Daemon.run(Daemons.java:111)
       at java.lang.Thread.run(Thread.java:764)

#16. Measurement Worker
       at java.lang.Object.wait(Object.java)
       at java.lang.Object.wait(Object.java:422)
       at com.google.android.gms.internal.measurement.zzgk.run(Unknown Source:76)
```

**Comment by pawelDylag** [comment]
Should be fixed in `0.6.0` :)
